app.js

```diff
@@ -136,6 +137,32 @@ function addEventListener(indicator, emitter, event, f) {
   });
 }
 
+/*
+  Check couchDbUrl is a valid database URL.
+  @param {string} couchDbUrl - Database URL
+  @param {function(err)} callback - error is undefined if DB exists
+*/
+function exists(db, isBackup, callback) {
+  db.head('', function(err) {
+    err = error.convertResponseError(err, function(err) {
```

Comment:
It seems to me that this method would be cleaner if it just called `callback` with `true` or `false`, then the callee can construct the appropriate error message and you don't need the `isBackup` flag.

However I'm also happy for you to leave as-is, since calling a callback with an error object is probably more nodejs-idiomatic

Reply:
I see what you mean, the problem is slightly around the range of errors that can be received and constructing the error message with extra information. I didn't really want to have two errorFactory functions that effectively did the same 404 check, but just created a slightly different message. I'll see if I can think of a better way.

Reply:
I think part of the problem here is the naming `exists` makes it sound very simple, but it's more a case of `tryToReachTheDbAndCallbackWithAnErrorIfThereIsSomeProblemAndWeCannotProceed` - that seems a bit longwinded so I've tried `proceedIfDbValid`.
I've also moved the backup/restore message append into the respective functions to avoid the boolean passing.
38051ce5cedc7b2b0e49284143e50fcd301654e4 WDYT?